### PR TITLE
Stable train map

### PR DESCRIPTION
**Pull request by benjamin-cates**
Calculate training map at each logging frequency rather than on each batch. Hopefully this means the class-wise map will be more accurate

### CODE REVIEW

File: pyha_analyzer/train.py [hunk 1]
```diff
@@ -122,23 +122,22 @@ def train(model: TimmModel,
             if scheduler is not None:
                 scheduler.step()
 
-        log_pred = F.sigmoid(outputs)
-        dataset: PyhaDFDataset = data_loader.dataset # type: ignore
-        cmap, smap = map_metric(log_pred, labels, dataset.class_dist)
-        log_cmap += cmap
-        log_smap += smap
+        log_pred.append(F.sigmoid(outputs))
```
Comment (Sean1572) [inline pyha_analyzer/train.py]: Make sure to detach and put tensors on cpu. Otherwise this is a pusedo-memory leak on gpu since we never freed that memeory on gpu.

**Review by Sean1572 — CHANGES_REQUESTED** [review]
See changes about gpu memory concerns

File: pyha_analyzer/train.py (hunk 1 above)
Comment (benjamin-cates) [reply pyha_analyzer/train.py]: oooops

**Review by Sean1572 — APPROVED** [review]
